### PR TITLE
S3 resource documentation

### PR DESCRIPTION
**Pull request by fhennig**
This is part of ticket #187

### CODE REVIEW

File: modules/concepts/pages/s3.adoc [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            inline: # <2>
+              host: test-minio
+              port: 9000
+  # more spec here ...
+----
+<1> The inline definition of the bucket. The bucket definition contains `bucketName` and `connection`.
+<2> The inline definition of the connection. It contains the `host` and `port`.
+
+
+==== Dedicated resources
```
Comment (razvan) [inline modules/concepts/pages/s3.adoc:110]: "Dedicated" doesn't sound like it separates them from inlined resources very well.  Maybe "free", "unwrapped" or "stand-alone" ?

File: modules/concepts/pages/s3.adoc [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  deepStorage:
+      s3:
+        reference: my-bucket-resource
+  # more spec here ...
+----
+
+== Credentials
+
+
+No matter if a connection is specified inline or as a separate object, the credentials are always specified in the same way. You will need a `Secret` containing the access key ID and secret access key, a `SecretClass` and then a reference to this `SecretClass` where you want to specify the credentials.
```
Comment (razvan) [inline modules/concepts/pages/s3.adoc:168]: I feel an explanation is need  to "why" a SecretClass is necessary? Why can't I just reference the Secret directly ?

File: modules/concepts/pages/s3.adoc (hunk 1 above)
Comment (fhennig) [reply modules/concepts/pages/s3.adoc:110]: Good point, I've changed it to stand-alone

**Review by razvan — APPROVED** [review]
SecretClass clarifications will follow in a new PR

**Pull request merged**